src/plugins/explore/public/components/visualizations/state_timeline/state_timeline_utils.ts

```diff
@@ -0,0 +1,546 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { groupBy } from 'lodash';
+import { RangeValue, Threshold, ValueMapping } from '../types';
+
+const addThresholdTime = (currentTime: string, threshold: string): number | undefined => {
+  const date = new Date(currentTime.replace(' ', 'T'));
+  if (isNaN(date.getTime())) {
+    return undefined;
+  }
+
+  const match = threshold.match(/^(\d+)([hms])$/);
+  if (!match) {
+    return undefined;
+  }
+
+  const value = Number(match[1]);
+  const unit = match[2];
+  let offset = 0;
+
+  switch (unit) {
+    case 'h':
+      offset = value * 60 * 60 * 1000;
+      break;
+    case 'm':
+      offset = value * 60 * 1000;
+      break;
+    case 's':
+      offset = value * 1000;
+      break;
+  }
+
+  return date.getTime() + offset;
+};
+
+const disconnectValues = (next: string, last: string, disconnectThreshold?: string): string => {
```

Comment:
Nit: Need to update the naming for this function, something like `generatedDisconnectTimestamp`.